packages/coding-agent/src/core/agent-session.ts

```diff
@@ -717,26 +827,72 @@ export class AgentSession {
 			}
 
 			const entries = this.sessionManager.loadEntries();
-			const compactionEntry = await compact(
-				entries,
-				this.model,
-				settings,
-				apiKey,
-				this._autoCompactionAbortController.signal,
-			);
+			let compactionEntry: CompactionEntry;
+			let usedCache = false;
```

Comment:
This duplicates the code in compact() above for the most part afaict. Please consolidate.